src/promptflow/tests/executor/unittests/batch/test_chat_group_orchestrator.py

```diff
@@ -0,0 +1,222 @@
+            chat_role = self._chat_group_roles[role_index]
+            chat_role_input = batch_inputs[role_index]
+            conversation_history_key = next(
+                (key for key, value in chat_role.inputs_mapping.items()
+                 if value == CONVERSATION_HISTORY_EXPRESSION), None
+            )
+            if conversation_history_key is None:
+                bulk_logger.error(
+                    f"Cannot find conversation expression mapping for "
+                    f"chat role: {chat_role.role}. name: {chat_role.name}"
+                )
+                message = (
+                    f"Cannot find conversation expression mapping for "
+                    f"chat role: {chat_role.role}. name: {chat_role.name} "
+                    f"Please use define {CONVERSATION_HISTORY_EXPRESSION} for a flow input."
+                )
+                raise MissingConversationHistoryExpression(message=message)
+            chat_role_input[conversation_history_key] = conversation_history
+            bulk_logger.info(
+                f"Start to execute turn {turn}. role: {chat_role.role}. name: {chat_role.name}"
+            )
+
+            current_line_result = await executor_proxy.exec_line_async(chat_role_input, line_index, run_id)
+            self._process_flow_outputs(
+                turn,
+                chat_role,
+                current_line_result,
+                conversation_history,
+                outputs,
+                aggregation_inputs)
+            bulk_logger.info(
+                f"Finish process line result for "
+                f"line number: {line_index}, turn:{turn}. role:{chat_role.role}, name: {chat_role.name}"
+            )
+
+            if any(value == chat_role.stop_signal for value in current_line_result.output.values()):
+                bulk_logger.info(
+                    f"Stop chat since current turn align with stop signal. "
+                    f"line number: {line_index}, turn:{turn}. role:{chat_role.role}, name: {chat_role.name}"
+                )
+                break
+
+        bulk_logger.info(
+            f"Finish schedule runs for run id: {run_id}, "
+            f"line number: {line_index}, add conversation history to output"
+        )
+        outputs.update({CONVERSATION_HISTORY_OUTPUT_KEY: conversation_history})
+
+        return LineResult(
+            output=outputs,
+            aggregation_inputs=aggregation_inputs,
+            node_run_infos=current_line_result.node_run_infos,
+            run_info=current_line_result.run_info
+        )
+
+    def _process_flow_outputs(
+            self,
+            index: int,
+            chat_role: ChatRole,
+            current_line_result: LineResult,
+            conversation_history: List[Mapping[str, Any]],
+            outputs: dict,
+            aggregation_inputs: dict):
+
+        current_turn = {"role": chat_role.role}
+        current_turn.update(current_line_result.output)
```

Comment:
Again, here will cause problem if `current_line_result.output` contains key with `role`

Reply:
We discussed this before offline, and intend to keep "role" as a reserved key.